database/README.md

```diff
@@ -1,14 +1,414 @@
-* the erc20 spending is approved
-* valid_to is not in the past and not more than <1 year> in the future
-* the order owner does not already have more than <1000> valid orders
+Some tables only store data emitted via smart contract events. Because we only have a single deployment of the [`GPv2Settlement`](https://github.com/cowprotocol/contracts/blob/main/src/contracts/GPv2Settlement.sol) settlement contract shared across staging and production environments events related to staging **and** production orders and settlements will be present in **both** the staging **and** production databases.  
+[CoWSwapEthFlow](https://github.com/cowprotocol/ethflowcontract/blob/main/src/CoWSwapEthFlow.sol) we actually deployed twice so events related to the staging environment should only show up in the staging DB and likewise for production.  
+It's also important to note that we only index events from blocks that we are certain will not get reorged. That means specifically that events will be indexed with a block delay of at least 64.  
 
-This ensures that the amount of orders we have to keep track of per user is limited and that there is a real cost the user incurs before being able to submit orders because they need to have sent an ethereum transaction.
+### auction\_participants
+
+This table is used for [CIP-20](https://snapshot.org/#/cow.eth/proposal/0x2d3f9bd1ea72dca84b03e97dda3efc1f4a42a772c54bd2037e8b62e7d09a491f). It stores which solvers (identified by ethereum address) participated in which auctions (identified by auction id). CIP-20 specifies that "solver teams which consistently provide solutions" get rewarded.
+
+   Column     |  Type  | Nullable | Details
+--------------|--------|----------|--------
+ auction\_id  | bigint | not null | id of the auction
+ participant  | bytea  | not null | solver that submitted a **valid** solution for the auction
+
+Indexes:  
+- PRIMARY KEY: btree(`auction_id`, `participant`)
+
+### auction\_prices
+
+Stores the native price of a token in a given auction. Used for computations related to CIP-20.
+
+ Column     | Type    | Nullable | Details
+------------|---------|----------|--------
+auction\_id | bigint  | not null | in which auction this price was provided
+token       | bytea   | not null | address of the token the price refers to
+price       | numeric | not null | the atoms of ETH that can be bought with 1 atom of the token
+
+Indexes:  
+- PRIMARY KEY: btree(`auction_uid`, `token`)  
+
+### auction\_transaction
+
+Because the the transaction hash of a given settlement depends on the gas parameters it ultimately gets submitted with onchain we can't know the hash before it got submitted. That's why we store the transaction sender (`tx_from`) and next nonce of the winning solver. With that information we can later associate the auction with the transaction that brought it onchain by cross-referencing the `tx_from` and `tx_nonce`.  
```

Comment:
```suggestion
Because the transaction hash of a given settlement depends on the gas parameters it ultimately gets submitted with onchain we can't know the hash before it got submitted. That's why we store the transaction sender (`tx_from`) and next nonce of the winning solver. With that information we can later associate the auction with the transaction that brought it onchain by cross-referencing the `tx_from` and `tx_nonce`.  
```